File: validatoryaml_old/src/main/java/com/validador/ui/settings/Configuration.java
package com.validador.ui.settings;

import java.io.*;
import java.util.Properties;

public class Configuration {
    private static final String CONFIG_DIR = System.getProperty("user.home") + File.separator + "Documentos" + File.separator + "ValidadorYAML";
    private static final String CONFIG_FILE = CONFIG_DIR + File.separator + "configuracion.properties";
    private static final String SPECTRAL_PATH_KEY = "spectralPath";
    private static final String LANGUAGE_KEY = "language";

    private String spectralPath;
    private String language = "es";

    public Configuration() {
        cargarConfiguracion();
    }

    private void cargarConfiguracion() {
        try {
            File dir = new File(CONFIG_DIR);
            if (!dir.exists()) dir.mkdirs();
            File file = new File(CONFIG_FILE);
            if (file.exists()) {
                Properties props = new Properties();
                try (FileInputStream fis = new FileInputStream(file)) {
                    props.load(fis);
                    spectralPath = props.getProperty(SPECTRAL_PATH_KEY, "");
                    language = props.getProperty(LANGUAGE_KEY, "es");
                }
            } else {
                spectralPath = "";
                language = "es";
            }
        } catch (Exception e) {
            spectralPath = "";
            language = "es";
        }
    }

    public boolean isSpectralPathSet() {
        return spectralPath != null && !spectralPath.trim().isEmpty();
    }

    public String getSpectralPath() {
        return spectralPath;
    }

    public void setSpectralPath(String path) {
        this.spectralPath = path;
        guardarConfiguracion();
    }

    public String getLanguage() {
        return language;
    }

    public void setLanguage(String lang) {
        this.language = lang;
        guardarConfiguracion();
    }

    private void guardarConfiguracion() {
        try {
            Properties props = new Properties();
            props.setProperty(SPECTRAL_PATH_KEY, spectralPath);
            props.setProperty(LANGUAGE_KEY, language);
            File dir = new File(CONFIG_DIR);
            if (!dir.exists()) dir.mkdirs();
            try (FileOutputStream fos = new FileOutputStream(CONFIG_FILE)) {
                props.store(fos, "Configuración Validador YAML");
            }
        } catch (Exception ignored) {}
    }
}

File: composeApp/src/desktopMain/kotlin/org/queststudios/yamlvalidation/config/Configuration.kt
package org.queststudios.yamlvalidation.config

import java.io.File
import java.io.FileInputStream
import java.io.FileOutputStream
import java.util.Properties

class Configuration {
    private val configDir = System.getProperty("user.home") + File.separator + "Documentos" + File.separator + "ValidadorYAML"
    private val configFile = configDir + File.separator + "configuracion.properties"
    private val spectralPathKey = "spectralPath"
    private val languageKey = "language"
    private val dontShowSpectralDialogKey = "dontShowSpectralDialog"

    var spectralPath: String = ""
        private set
    var language: String = "es"
        private set
    var dontShowSpectralDialog: Boolean = false
        private set

    init {
        loadConfig()
    }

    private fun loadConfig() {
        try {
            val dir = File(configDir)
            if (!dir.exists()) dir.mkdirs()
            val file = File(configFile)
            if (file.exists()) {
                val props = Properties()
                FileInputStream(file).use { fis -> props.load(fis) }
                spectralPath = props.getProperty(spectralPathKey, "")
                language = props.getProperty(languageKey, "es")
                dontShowSpectralDialog = props.getProperty(dontShowSpectralDialogKey, "false").toBoolean()
            } else {
                spectralPath = ""
                language = "es"
                dontShowSpectralDialog = false
            }
        } catch (_: Exception) {
            spectralPath = ""
            language = "es"
            dontShowSpectralDialog = false
        }
    }

    fun isSpectralPathSet(): Boolean = spectralPath.isNotBlank()

    fun setSpectralPath(path: String) {
        spectralPath = path
        saveConfig()
    }

    fun setLanguage(lang: String) {
        language = lang
        saveConfig()
    }

    fun setDontShowSpectralDialog(value: Boolean) {
        dontShowSpectralDialog = value
        saveConfig()
    }

    private fun saveConfig() {
        try {
            val props = Properties()
            props.setProperty(spectralPathKey, spectralPath)
            props.setProperty(languageKey, language)
            props.setProperty(dontShowSpectralDialogKey, dontShowSpectralDialog.toString())
            val dir = File(configDir)
            if (!dir.exists()) dir.mkdirs()
            FileOutputStream(configFile).use { fos -> props.store(fos, "Configuración Validador YAML") }
        } catch (_: Exception) {}
    }
}
